src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchResultsProcessor.kt

```diff
@@ -0,0 +1,104 @@
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PagePrisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchDataSource.GLOBAL_SEARCH
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchDataSource.MATCH_PRISONERS
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchType.NAME_DOB
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchType.PRISON_NUMBER
+
+@Component
+class PrisonerSearchResultsProcessor(
+  private val telemetryClient: TelemetryClient,
+  private val prisonerSearchResultsLogger: PrisonerSearchResultsLogger,
+) {
+
+  fun processSearchResults(
+    prisonerMatches: PrisonerMatches,
+    prisonerSearchRequest: PrisonerSearchRequest,
+    caseload: String
+  ) {
+    prisonerSearchResultsLogger.logResults(prisonerMatches, prisonerSearchRequest, caseload)
+
+    val prisoners = prisonerMatches.matches.map { it.prisoner }
+    val prisonerMatchCount = PrisonerMatchCount.of(prisoners, prisonerSearchRequest)
+    val customDimensions = PrisonerSearchCustomDimensions(
+      dataSource = MATCH_PRISONERS,
+      searchType = prisonerSearchRequest.searchType,
+      numberOfResults = prisoners.size,
+      exactMatchCount = prisonerMatchCount.mainDetailsCount,
+      aliasExactMatchCount = prisonerMatchCount.aliasDetailsCount
+    ).asMap()
+
+    telemetryClient.trackEvent("prisoner-search", customDimensions, null)
+  }
+
+  fun processSearchResults(
+    pagePrisoner: PagePrisoner,
+    prisonerSearchRequest: PrisonerSearchRequest,
+    caseload: String
+  ) {
+    prisonerSearchResultsLogger.logResults(pagePrisoner, prisonerSearchRequest, caseload)
+
+    val prisoners = pagePrisoner.content ?: emptyList()
+    val prisonerMatchCount = PrisonerMatchCount.of(prisoners, prisonerSearchRequest)
+    val customDimensions = PrisonerSearchCustomDimensions(
+      dataSource = GLOBAL_SEARCH,
+      searchType = prisonerSearchRequest.searchType,
+      numberOfResults = prisoners.size,
+      exactMatchCount = prisonerMatchCount.mainDetailsCount,
+      aliasExactMatchCount = prisonerMatchCount.aliasDetailsCount
+    ).asMap()
+
+    telemetryClient.trackEvent("prisoner-search", customDimensions, null)
+  }
+}
+
+private val PrisonerSearchRequest.searchType: PrisonerSearchType
+  get() = if (prisonNumber != null) PRISON_NUMBER else NAME_DOB
+
+data class PrisonerMatchCount(
+  val mainDetailsCount: Int,
+  val aliasDetailsCount: Int
+) {
+  companion object {
+    fun of(prisoners: List<Prisoner>, prisonerSearchRequest: PrisonerSearchRequest): PrisonerMatchCount {
```

Comment:
This class, and specifically this `of` method is thoroughly covered in unit tests 👍